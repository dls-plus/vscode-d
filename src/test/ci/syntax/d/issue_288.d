void main() {
	synchronized int x = 4;
	synchronized stateInfo.status = "can't connect (%s)".format(addr);
	return;
	synchronized (foo) {}
	int x;
}

synchronized class Foo {
	synchronized SomeFunction!(int) foo() {}
	synchronized int foo();
}