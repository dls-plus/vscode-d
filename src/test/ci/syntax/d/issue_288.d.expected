[{"start":0,"end":4,"scope":"storage.type.basic-type.d"},{"start":4,"end":5,"scope":"source.d"},{"start":5,"end":9,"scope":"entity.name.function.d"},{"start":9,"end":10,"scope":"source.d"},{"start":10,"end":11,"scope":"source.d"},{"start":11,"end":14,"scope":"source.d"}]
[{"start":0,"end":1,"scope":"source.d"},{"start":1,"end":13,"scope":"storage.class.d"},{"start":13,"end":14,"scope":"source.d"},{"start":14,"end":17,"scope":"storage.type.basic-type.d"},{"start":17,"end":18,"scope":"source.d"},{"start":18,"end":19,"scope":"variable.d"},{"start":19,"end":20,"scope":"source.d"},{"start":20,"end":21,"scope":"keyword.operator.assign.d"},{"start":21,"end":22,"scope":"source.d"},{"start":22,"end":23,"scope":"constant.numeric.integer.decimal.d"},{"start":23,"end":25,"scope":"meta.statement.end.d"}]
[{"start":0,"end":1,"scope":"source.d"},{"start":1,"end":13,"scope":"storage.class.d"},{"start":13,"end":14,"scope":"source.d"},{"start":14,"end":30,"scope":"variable.d"},{"start":30,"end":31,"scope":"source.d"},{"start":31,"end":32,"scope":"keyword.operator.assign.d"},{"start":32,"end":33,"scope":"source.d"},{"start":33,"end":34,"scope":"string.double-quoted-string.d"},{"start":34,"end":37,"scope":"string.character.d"},{"start":37,"end":38,"scope":"string.double-quoted-string.d"},{"start":38,"end":48,"scope":"string.character.d"},{"start":48,"end":50,"scope":"string.double-quoted-string.d"},{"start":50,"end":51,"scope":"string.character.d"},{"start":51,"end":52,"scope":"string.double-quoted-string.d"},{"start":52,"end":53,"scope":"string.double-quoted-string.d"},{"start":53,"end":60,"scope":"entity.name.function.d"},{"start":60,"end":61,"scope":"source.d"},{"start":61,"end":65,"scope":"variable.d"},{"start":65,"end":66,"scope":"source.d"},{"start":66,"end":68,"scope":"meta.statement.end.d"}]
[{"start":0,"end":1,"scope":"source.d"},{"start":1,"end":7,"scope":"keyword.control.return.d"},{"start":7,"end":9,"scope":"meta.statement.end.d"}]
[{"start":0,"end":1,"scope":"source.d"},{"start":1,"end":13,"scope":"keyword.control.synchronized.d"},{"start":13,"end":14,"scope":"source.d"},{"start":14,"end":15,"scope":"source.d"},{"start":15,"end":18,"scope":"variable.d"},{"start":18,"end":19,"scope":"source.d"},{"start":19,"end":23,"scope":"source.d"}]
[{"start":0,"end":1,"scope":"source.d"},{"start":1,"end":4,"scope":"storage.type.basic-type.d"},{"start":4,"end":5,"scope":"source.d"},{"start":5,"end":6,"scope":"variable.d"},{"start":6,"end":8,"scope":"meta.statement.end.d"}]
[{"start":0,"end":2,"scope":"source.d"}]
[{"start":0,"end":1,"scope":"source.d"}]
[{"start":0,"end":12,"scope":"storage.class.d"},{"start":12,"end":13,"scope":"source.d"},{"start":13,"end":18,"scope":"storage.type.class.d"},{"start":18,"end":19,"scope":"source.d"},{"start":19,"end":22,"scope":"entity.name.class.d"},{"start":22,"end":25,"scope":"source.d"}]
[{"start":0,"end":1,"scope":"source.d"},{"start":1,"end":13,"scope":"storage.class.d"},{"start":13,"end":14,"scope":"source.d"},{"start":14,"end":26,"scope":"variable.d"},{"start":26,"end":27,"scope":"keyword.operator.logical.d"},{"start":27,"end":28,"scope":"source.d"},{"start":28,"end":31,"scope":"storage.type.basic-type.d"},{"start":31,"end":32,"scope":"source.d"},{"start":32,"end":33,"scope":"source.d"},{"start":33,"end":36,"scope":"entity.name.function.d"},{"start":36,"end":37,"scope":"source.d"},{"start":37,"end":38,"scope":"source.d"},{"start":38,"end":42,"scope":"source.d"}]
[{"start":0,"end":1,"scope":"source.d"},{"start":1,"end":13,"scope":"storage.class.d"},{"start":13,"end":14,"scope":"source.d"},{"start":14,"end":17,"scope":"storage.type.basic-type.d"},{"start":17,"end":18,"scope":"source.d"},{"start":18,"end":21,"scope":"entity.name.function.d"},{"start":21,"end":22,"scope":"source.d"},{"start":22,"end":23,"scope":"source.d"},{"start":23,"end":25,"scope":"meta.statement.end.d"}]
[{"start":0,"end":2,"scope":"source.d"}]